builtins/src/types/numbers.rs

```diff
@@ -61,4 +114,212 @@ mod tests {
+            0xFFFF_FFFF_FFFF_FFFEu64 as f64,
+            0xFFF0_0000_0000_0001u64 as f64,
+            0.0,
+            -0.0,
+            2.3,
+            23.0,
+            230.0,
+            2300.0,
+            23000.0,
+            230000.0,
+            23e5,
+            23e6,
+            2.3e5,
+            0.23,
+            0.023,
+            0.0023,
+            0.00023,
+            0.000023,
+            0.0000023,
+            0.23e-5,
+            f32::MIN_POSITIVE as f64, // 42
+            f32::MIN_POSITIVE as f64 / 3.0,
+            f32::MIN_POSITIVE as f64 / 7e5,
+            f32::MIN_POSITIVE as f64 / 7e6,
+            f32::MIN_POSITIVE as f64 / 7e7,
+            f32::MIN_POSITIVE as f64 / 7e8,
+            f32::MIN_POSITIVE as f64 / 4.123e14,
+            f32::MAX as f64,
+            f32::MAX as f64 * 3.0,
+            f32::MAX as f64 + 99.0,
+            f64::MIN_POSITIVE,
+            f64::MIN_POSITIVE / 2.0,
+            f64::MIN_POSITIVE / 10.0,
+            f64::MIN,
+            f64::MAX,
+            f64::INFINITY,
+            f64::NEG_INFINITY,
+            f64::NAN,
+            F56::EPSILON,
+            F56::EPSILON / 3.0,
+            f64::from(F56::MAX),
+            f64::from(F56::MAX) + 1.0,
+            f64::from(F56::MIN_POSITIVE),
+            f64::from(F56::MIN_POSITIVE) / 3.0, // #55
+            f64::from(F56::MIN_POSITIVE_SUBNORMAL),
+            f64::from(F56::MIN_POSITIVE_SUBNORMAL) / 3.0,
+            -f64::from(F56::MAX),
+            -f64::from(F56::MAX) - 1.0,
+            -f64::from(F56::MAX) * 3.3,
+            8.4e-168,
+            8.4e-169,
+            8.4e-170,
+        ];
+
+        fn debug(orig_f64: f64, index: usize) {
+            println!("index: {}", index);
+            println!("original f64      : {}", log_f64(orig_f64));
+            println!("f64 -> f32        : {}", log_f32(orig_f64 as f32));
+            println!("f64 -> f32 -> f64 : {}", log_f64((orig_f64 as f32) as f64));
+            println!("f64 -> f56        : {}", log_f56(F56::from(orig_f64)));
+            println!(
+                "f64 -> f56 -> f64 : {}",
+                log_f64(f64::from(F56::from(orig_f64)))
+            );
+            let f32_diff = (f64::from(orig_f64 as f32) - orig_f64).abs();
+            let f32_relative_difference = f32_diff / orig_f64.abs();
+            if f32_relative_difference > 0.0 {
+                println!("f32 relative difference {:.5e}", f32_relative_difference);
+            }
+            let f56_diff: f64 = (f64::from(F56::from(orig_f64)) - orig_f64).abs();
+            let f56_relative_difference = f56_diff / orig_f64.abs();
+            if f56_relative_difference > 0.0 {
+                println!("f56 relative difference {:.5e}", f56_relative_difference);
+            }
+            println!("")
+        }
+
+        for index in 0..numbers_to_test.len() {
+            let orig_f64 = numbers_to_test[index];
+            /* Calculate the values  */
```

Comment:
for comments like these within functions, use the double slash... even if they're multiline, which I know is annoying.
```
// this comment is about assiging foo
// and can extend multiple lines.
let foo = 0.0;
```